src/v/security/audit/probes.cc

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "config/configuration.h"
+#include "prometheus/prometheus_sanitize.h"
+#include "security/audit/audit_log_manager.h"
+#include "security/audit/client_probe.h"
+#include "security/audit/probe.h"
+
+#include <seastar/core/metrics.hh>
+
+#include <chrono>
+
+namespace security::audit {
+
+void audit_probe::setup_metrics(std::function<double()> get_usage_ratio) {
```

Comment:
Two notes here:

You are using `ssx::metrics::all_metrics_group`. This will expose the metrics on internal metrics and public metrics (these are the ones that customers have access to in cloudv2) by default. On internal metrics you'd obviously want all of them but possibly you only want a subset (or none at all) on public metrics if they are not useful to a customer (for example is the buffer usage ratio useful to them?).

Another point is that `aggregate_metrics` generally only applies to internal metrics. We want to keep public metrics as small as possible. Hence on public metrics we do always aggregate the shard label (and this is also what drives the subset consideration from above). Assuming you want to expose all of them on internal and public metrics (and not do a subset) you can use a pattern like [this](https://github.com/redpanda-data/redpanda/blob/dev/src/v/raft/recovery_scheduler.cc#L365) to avoid specifying everything twice but apply different aggregation logic.

Above applies to `client_probe` in the same vein.

Reply:
Exposing these metrics on both endpoints was intentional, yeah. 

>aggregate_metrics generally only applies to internal metrics

Cool, I hadn't grokked that. Will update to follow the linked pattern. Should make things a bit easier if we add metrics that _should_ be exclusively internal in the future.

Reply:
thanks @StephanDollberg TIL